File: src/main/java/com/contrader/contraininggame/service/Service.java
package com.contrader.contraininggame.service;

import java.util.Optional;

public interface Service<Model, KeyType> {
    Iterable<Model> getAll();

    Optional<Model> read(KeyType id);

    Model insert (Model dto);

    Model update (Model dto);

    void delete (KeyType id);

}

File: src/service/service.ts
import { Observable } from 'rxjs';

export interface Service<DTO>{
    getAll(): Observable<DTO[]>;
    read(id: number): Observable<DTO>;
    insert(data:DTO): Observable<any>;
    update(data:DTO): Observable<any>;
    delete(id:number): Observable<any>;
}
